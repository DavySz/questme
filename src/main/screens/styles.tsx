import { GestureHandlerRootView } from "react-native-gesture-handler";
import styled from "styled-components/native";

export const GestureWrapper = styled(GestureHandlerRootView)`
  flex: 1;
`;
