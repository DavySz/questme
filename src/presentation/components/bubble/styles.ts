import styled, { css } from "styled-components/native";
import { ICircleStyles, IPositionWrapperStyles } from "./types";

export const getWrapperPosition = (
  position: Partial<IPositionWrapperStyles>
) => {
  const customPosition = [];

  if (position.bottom) {
    customPosition.push(css`
      bottom: ${position.bottom}px;
    `);
  }

  if (position.left) {
    customPosition.push(css`
      left: ${position.left}px;
    `);
  }

  if (position.top) {
    customPosition.push(css`
      top: ${position.top}px;
    `);
  }

  if (position.right) {
    customPosition.push(css`
      right: ${position.right}px;
    `);
  }

  return customPosition;
};

export const CircleFill = styled.View<ICircleStyles>`
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;

  border-radius: ${({ size }) => size / 2}px;

  background-color: ${({ theme }) => theme.colors.support["accent-2"]};

  opacity: 0.1;
`;

export const CircleOutline = styled.View<ICircleStyles>`
  width: ${({ size }) => size}px;
  height: ${({ size }) => size}px;

  border-radius: ${({ size }) => size / 2}px;

  background-color: ${({ theme }) => theme.colors.transparent};
  border: 1px solid ${({ theme }) => theme.colors.support["accent-2"]};

  opacity: 0.1;
`;

export const PositionWrapper = styled.View<Partial<IPositionWrapperStyles>>`
  position: absolute;
  ${(props) => getWrapperPosition(props as Partial<IPositionWrapperStyles>)};
`;
