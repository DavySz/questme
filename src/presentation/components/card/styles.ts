import styled from "styled-components/native";

export const Wrapper = styled.View`
  flex-direction: row;

  align-items: center;
  justify-content: space-between;

  width: 100%;
  padding: 8px;

  border-radius: 20px;
  border: 2px solid ${({ theme }) => theme.colors.neutral["grey-5"]};

  background-color: ${({ theme }) => theme.colors.neutral.white};
`;

export const Content = styled.View`
  gap: 6px;
`;

export const Group = styled.View`
  gap: 6px;
`;
