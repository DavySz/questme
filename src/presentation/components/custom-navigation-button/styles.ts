import styled from "styled-components/native";

export const Wrapper = styled.TouchableOpacity`
  border-radius: 24px;
  height: 48px;
  width: 48px;

  justify-content: center;
  align-items: center;

  margin-top: -20px;
`;

export const Container = styled.View`
  border-radius: 32px;
  height: 64px;
  width: 64px;

  justify-content: center;
  align-items: center;

  background-color: ${({ theme }) => theme.colors.transparent};

  margin-top: -20px;
`;
