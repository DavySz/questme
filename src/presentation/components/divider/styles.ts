import styled from "styled-components/native";

export const Line = styled.View`
  width: 100%;
  height: 1px;
  flex-direction: row;
  background-color: ${({ theme }) => theme.colors.neutral["gey-4"]};
  margin: 0 16px;
`;
