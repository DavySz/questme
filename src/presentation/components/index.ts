export * from "./button";
export * from "./text";
export * from "./screen-template";
export * from "./divider";
