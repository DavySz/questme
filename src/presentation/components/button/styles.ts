import styled, { css } from "styled-components/native";
import {
  IButtonIconStyles,
  IRootStyle,
  TButtonVariant,
  TIconSide,
} from "./types";
import { TTheme, TThemeColors } from "@presentation/styles";

export const getButtonBackground = (
  variant: TButtonVariant,
  theme: TTheme
): string => {
  const colors: Record<TButtonVariant, string> = {
    primary: theme.colors.primary,
    secondary: theme.colors.neutral["grey-5"],
    tertiary: theme.colors.transparent,
    facebook: theme.colors.company.facebook,
    google: theme.colors.neutral.white,
    location: theme.colors.neutral.white,
  };

  return colors[variant];
};

export const getButtonTextColor = (variant: TButtonVariant): TThemeColors => {
  const color: Record<TButtonVariant, TThemeColors> = {
    primary: "neutral-white",
    secondary: "primary",
    tertiary: "primary",
    facebook: "neutral-white",
    google: "neutral-black",
    location: "primary",
  };

  return color[variant] as TThemeColors;
};

const isFull = (isFull: boolean) => {
  if (isFull) {
    return css`
      width: 100%;
    `;
  }
};

const getIconSide = (side: TIconSide) => {
  if (side === "left") {
    return css`
      margin-right: 16px;
    `;
  }

  if (side === "right") {
    return css`
      margin-left: 16px;
    `;
  }

  return null;
};

const getBorder = (variant: TButtonVariant, theme: TTheme) => {
  if (variant === "google") {
    return css`
      border: 1px solid ${theme.colors.neutral["grey-4"]};
    `;
  }
};

export const Button = styled.TouchableOpacity.attrs({
  activeOpacity: 0.7,
})<IRootStyle>`
  ${({ full }) => isFull(full)};
  ${({ variant, theme }) => getBorder(variant, theme as TTheme)};

  background-color: ${({ theme, variant }) =>
    getButtonBackground(variant, theme as TTheme)};

  border-radius: 20px;
  padding: 16px;

  flex-direction: row;

  align-items: center;
  justify-content: center;
`;

export const IconWrapper = styled.View<IButtonIconStyles>`
  ${({ side }) => getIconSide(side)}
`;
