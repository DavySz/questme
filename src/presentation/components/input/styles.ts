import styled from "styled-components/native";
import { IInputWrapperStyles } from "./types";
import { TTheme } from "@presentation/styles";

const getBorderColor = (isFocused: boolean, theme: TTheme) => {
  if (isFocused) return theme.colors.primary;
  return theme.colors.neutral["grey-5"];
};

export const CustomInput = styled.TextInput`
  width: 100%;
  font-family: ${({ theme }) => theme.fonts.regular};
  font-size: 16px;
`;

export const InputWrapper = styled.View<IInputWrapperStyles>`
  flex-direction: row;

  width: 100%;
  padding: 16px;
  border-radius: 20px;

  gap: 16px;

  border: 2px solid
    ${({ isFocused, theme }) => getBorderColor(isFocused, theme as TTheme)};

  background-color: ${({ theme }) => theme.colors.neutral.white};
`;

export const Wrapper = styled.View`
  margin-bottom: 8px;
`;
