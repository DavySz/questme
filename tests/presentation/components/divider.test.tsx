import { Divider } from "@presentation/components";
import { theme } from "@presentation/styles";
import { render } from "@testing-library/react-native";
import { ThemeProvider } from "styled-components/native";

describe("Divider", () => {
  test("Should render component without errors", () => {
    const tree = render(
      <ThemeProvider theme={theme}>
        <Divider.Text>any-text</Divider.Text>
        <Divider.Root />
      </ThemeProvider>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
