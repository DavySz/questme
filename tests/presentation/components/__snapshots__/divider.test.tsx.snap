// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Should render component without errors 1`] = `
[
  <Text
    color="neutral-gey-2"
    style={
      {
        "color": "#858494",
        "fontFamily": "Rubik_400Regular",
        "fontSize": 16,
        "lineHeight": 24,
      }
    }
    variant="body-normal-regular"
  >
    any-text
  </Text>,
  <View
    style={
      {
        "backgroundColor": "#E6E6E6",
        "flexDirection": "row",
        "height": 1,
        "marginBottom": 0,
        "marginLeft": 16,
        "marginRight": 16,
        "marginTop": 0,
        "width": "100%",
      }
    }
  />,
]
`;
